src/flow/src/compute/render/reduce.rs

```diff
@@ -397,6 +401,24 @@ fn reduce_accum_subgraph(
     // TODO(discord9): consider key-based lock
     let mut arrange = arrange.write();
     for (key, value_diffs) in key_to_vals {
+        if let Some(expire_man) = &arrange.get_expire_state() {
+            let mut is_expired = false;
+            err_collector.run(|| {
+                if let Some(expired) = expire_man.get_expire_duration(now, &key)? {
+                    is_expired = true;
+                    DataAlreadyExpiredSnafu {
```

Comment:
Should late data arrive be collected as a error? Or separately be collected only by a distribution of time or just max time?

Reply:
Expired data should be ignored. Maybe a warning log is better.